chain-signatures/node/src/protocol/signature.rs

```diff
@@ -28,34 +27,91 @@ use tokio::sync::mpsc::error::TryRecvError;
 use tokio::sync::{mpsc, RwLock};
 
 use near_account_id::AccountId;
-use near_primitives::hash::CryptoHash;
-
-pub type ReceiptId = near_primitives::hash::CryptoHash;
 
 /// This is the maximum amount of sign requests that we can accept in the network.
 const MAX_SIGN_REQUESTS: usize = 1024;
 
-pub struct SignRequest {
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SignId {
+    #[serde(with = "serde_bytes")]
     pub request_id: [u8; 32],
-    pub request: ContractSignRequest,
+    #[serde(with = "cbor_scalar")]
+    pub payload: Scalar,
+    #[serde(with = "cbor_scalar")]
     pub epsilon: Scalar,
+}
+
+impl std::hash::Hash for SignId {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.request_id.hash(state);
+        self.payload.to_bytes().hash(state);
+        self.epsilon.to_bytes().hash(state);
+    }
+}
+
+impl std::fmt::Debug for SignId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("SignId")
+            .field(&near_primitives::hash::CryptoHash(self.request_id))
+            .field(&hex::encode(self.payload.to_bytes()))
+            .field(&hex::encode(self.epsilon.to_bytes()))
+            .finish()
+    }
+}
+
+impl SignId {
+    pub fn new(request_id: [u8; 32], epsilon: Scalar, payload: Scalar) -> Self {
+        Self {
+            request_id,
+            epsilon,
+            payload,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SignArgs {
+    #[serde(with = "serde_bytes")]
     pub entropy: [u8; 32],
-    pub time_added: Instant,
+    pub path: String,
```

Comment:
Why do we need to pass the path around, if we already have epsilon in SignId? My suggestion is still to remove epsilon and payload from SignId.

Reply:
we can't recover the path from an epsilon, so better to just keep it in our system for now